CIP-0062/README.md

```diff
@@ -0,0 +1,192 @@
+
+### **Returns**
+cbor hex encoded representation of the public key.
+
+## **api.submitDelegation(delegation: Delegation): Promise\<SignedDelegationMetadata>**
+
+This endpoint should construct the cbor encoded delegation certificate according to the specs in [CIP-36 Example](https://github.com/Zeegomo/CIPs/blob/472181b9c69feeedae0b5b2db8b42d0cf4eb1a11/CIP-0036/README.md#example).
+
+It should then sign the certificate with the staking key as described in the same example as above.
+
+The implementation of this endpoint can make use of the already existing [CIP-30](https://cips.cardano.org/cips/cip30/) `api.signData` and `api.submitTx` to perform the broadcasting of the transaction containing the metadata. 
+
+Upon submission of the transaction containing the delegation cert as part of metadata, the wallet should store the delegation in its local storage and return an object that contains the delegation cert, the signature and the txhash of the transaction that the certificate was submitted with.
+
+## **`Delegation`**
+
+```
+export interface Delegation {
+    delegations: GovernanceKey[],
+    purpose: Purpose
+}
+```
+
+Defines the structure to be crafted and signed for delegation of voting & their respectively voting power.   Embeds the stake key and reward address from the wallet, and constructs a suitable nonce.
+
+***`delegations`***: List of keys and their voting weight to delegate voting power to.
+
+This should be a call that implicitly cbor encodes the `delegation` object and uses the already existing [CIP-30](https://cips.cardano.org/cips/cip30/) `api.submitTx` to submit the transaction. The resulting transaction hash should be returned.
+
+This should be trigger a request to the wallet to approve the transaction.
+
+### **Returns**
+
+```
+interface SignedDelegationMetadata {
+  certificate: DelegatedCertificate,
+  signature: string, // signature on the CIP-36 delegation certificate
+  txHash: string // of the transaction that submitted the delegation 
+}
+```
+
+#### **DelegatedCertificate**
+```
+interface DelegatedCertificate {
+  delegations: GovernanceKey[] // key delegations,
+  stakingPub: string // staking public key
+  rewardAddress: string // reward address
+  nonce: number //nonce used in the transaction
+  purpose: Purpose
+}
+```
+
+Errors: `APIError`, `TxSendError`
+
+## **Delegation Cert process**
+
+1. **`Get Voting Key`** - use the method **api.getVotingKey** to return a ed25519 32 bytes public key (x value of the point on the curve).
+
+2. **`Collect Voting Keys`** - Collect the keys to delegate voting power to.
+
+3. **`Submit delegation`** - Submit the metadata transaction to the chain using **api.submitDelegation** which implicitly sign and/or can use the already existing **api.submitTx**,  available from [CIP-30](https://cips.cardano.org/cips/cip30/)
```

Comment:
```suggestion
3. **`Submit delegation`** - Submit the metadata transaction to the chain using **api.submitDelegation** which implicitly signs it
```
I think with the current CIP-0062 API it's not really possible/reasonable to use `api.submitTx` to submit the delegation transaction. The `api.submitDelegation` call doesn't return back the full signed registration transaction and even if it did, it seems counterproductive to submit the transaction again if the `submitDelegation` call is already required to do that, I'd even expect most implementation to await the tx being submitted before returning, making further submission by `api.submitTx` pointless

Reply:
Clearing this, as discussed that we only support signing votes through this API and the dapp should be able to submit it.